Foundation/JSONSerialization.swift

```diff
@@ -812,6 +812,41 @@ private struct JSONReader {
         0x2E, 0x2D, 0x2B, 0x45, 0x65, // . - + E e
     ]
     func parseNumber(_ input: Index, options opt: JSONSerialization.ReadingOptions) throws -> (Any, Index)? {
+        #if !arch(i386) && !arch(arm) // x64
+            func castInt64(_ value: Int64) -> Any {
```

Comment:
Its better to use `Int.bitWidth == 64` rather than adding a list of architectures which would need updating

Reply:
I agree this looks ugly, but I want to move this to compile time. Do you know some better way to make a check like this without extra runtime cost?

Reply:
Do you expect more 32bit arch in future? I think all new archs would be 64bit and this if will be work correctly.

Reply:
The compiler knows how to optimize `Int.bitWidth == 64` to a compile-time constant.